sklearn/feature_extraction/text.py

```diff
@@ -676,12 +682,8 @@ def __init__(self, input='content', encoding='utf-8',
         if max_df < 0 or min_df < 0:
             raise ValueError("negative value for max_df or min_df")
         self.max_features = max_features
-        if max_features is not None:
-            if (not isinstance(max_features, numbers.Integral) or
-                    max_features <= 0):
-                raise ValueError(
-                    "max_features=%r, neither a positive integer nor None"
-                    % max_features)
+        if max_features is not None and max_features < 0:
+            raise ValueError("negative value for max_features")
```

Comment:
Fine, keep it like this for now but it would be great to improve the type checking of `mi_df`, `max_df` and `max_features` as well as the error messages in a separate PR.